pkg/utils/nodes.go

```diff
@@ -0,0 +1,28 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IsNodeNameValid returns an error if nodeName doesn't match any node name int the cluster, otherwise a nil
+func IsNodeNameValid(r client.Reader, nodeName string) (bool, error) {
+	node := &corev1.Node{}
+	nodeExist := true
```

Comment:
what do you need this for? IMHO it would be easier to read when using true or false directly
```
if isNotFound(err) {
  return false, nil
} else if err != nil {
  return false, err
}
return true, nil
```
or if you want to keep error handling nested:
```
if err != nil {
  if isNotFound(err) {
    return false, nil
  }
  return false, err
}
return true, nil
```